### PR TITLE
Adds quick_discouraged_pending_test opt-in rule

### PR DESCRIPTION
**Pull request by ornithocoder**
Implements #1909.

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/QuickDiscouragedPendingTestRule.swift [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                               severity: configuration.severity,
+                               location: Location(file: file, byteOffset: offset))]
+    }
+}
+
+private enum QuickPendingCallKind: String {
+    case pending
+    case xdescribe
+    case xcontext
+    case xit
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/QuickDiscouragedPendingTestRule.swift]: probably `xitBehavesLike` should be here too

Comment (ornithocoder) [reply Source/SwiftLintFramework/Rules/QuickDiscouragedPendingTestRule.swift]: Interesting. Quick's SDL.swift doesn't have `xitBehavesLike`. It's defined as a #define in QCKSDL.h. I'll add it.

**Comment by ornithocoder** [comment]
That's weird. GitHub didn't update the PR with the commits from my branch.

**Comment by ornithocoder** [comment]
I guess I'll have to close this one and open another PR. GitHub didn't pick my last two commits.

**Pull request closed without merging**